main.tf

```diff
@@ -178,7 +178,8 @@ resource "aws_codedeploy_deployment_group" "default" {
   # Note that you cannot have both ec_tag_filter and ec2_tag_set vars set!
   # See https://docs.aws.amazon.com/cli/latest/reference/deploy/create-deployment-group.html for details
   dynamic "ec2_tag_filter" {
-    for_each = length(var.ec2_tag_filter) > 0 ? [] : var.ec2_tag_filter
```

Comment:
if the length is greater than 0, then it should use var.ec2_tag_filter. in the deleted logic, it was not used if it was greater than zero.

Reply:
Oh I see. Good grief. Literally the opposite of what we wanted. I feel a bit dumb now